pkg/storage/redis/redis.go

```diff
@@ -66,13 +68,18 @@ func New(address, recordType string, deletePermanentAfter int64) (*DB, error) {
 		deletedSet:             recordType + "_deleted_set",
 		lastVersionKey:         recordType + "_last_version",
 	}
+	metrics.AddRedisMetrics(db.pool.Stats)
 	return db, nil
 }
 
 // Put sets new record for given id with input data.
-func (db *DB) Put(ctx context.Context, id string, data *anypb.Any) error {
+func (db *DB) Put(ctx context.Context, id string, data *anypb.Any) (err error) {
 	c := db.pool.Get()
+	_, span := trace.StartSpan(ctx, "databroker.redis.Put")
+	defer span.End()
```

Comment:
Three back-to-back defer statements feels like a code smell to me. E.g. Should we be calling `defer c.Close()` here before starting the trace span?

Reply:
Yeah, maybe it's just me but I'm having a really hard time keeping tack of how the defer statements are going to unwind if there's an error somewhere.

Reply:
Could change to this:

```go
	defer func() {
		c.Close()
		recordOperation(ctx, time.Now(), "put", err)
		span.End()
	}()
```

Reply:
I'm not sure that does the same thing though. 🤔 
\cc @cuonglm  / @calebdoxsey thoughts?

Reply:
defers are called in reverse order:
```go
defer a()
defer b()
```
runs b() then a()

Reply:
Correction, it would be:

```go
	defer func(s time.Time) {
		c.Close()
		recordOperation(ctx, time.Since(s), "put", err)
		span.End()
	}(time.Now())
```

@desimone any opinion on which style?